rest/config.go

```diff
@@ -266,6 +267,55 @@ func GetTLSVersionFromString(stringV *string) uint16 {
 	return uint16(DefaultMinimumTLSVersionConst)
 }
 
+type invalidConfigInfo struct {
+	logged              bool
+	configBucketName    string
+	persistedBucketName string
+}
+
+type invalidDatabaseConfigs struct {
+	dbNames    map[string]*invalidConfigInfo
+	loggingCtx context.Context
```

Comment:
Storing a context on a struct is an anti-pattern https://go.dev/blog/context-and-structs so we are trying to fix this as we go, let's pass this into `addInvalidDatabase`

In addition to confusion, the biggest problem is that we can easily pass `nil` here.